prow/scripts/cluster-integration/kyma-integration-gardener-eventing-upgrade.sh

```diff
@@ -0,0 +1,169 @@
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/gcp.sh"
+else
+    log::error "GARDENER_PROVIDER ${GARDENER_PROVIDER} is not yet supported"
+    exit 1
+fi
+
+# nice cleanup on exit, be it successful or on fail
+trap gardener::cleanup EXIT INT
+
+#Used to detect errors for logging purposes
+ERROR_LOGGING_GUARD="true"
+export ERROR_LOGGING_GUARD
+
+RANDOM_NAME_SUFFIX=$(LC_ALL=C tr -dc 'a-z0-9' < /dev/urandom | head -c6)
+readonly COMMON_NAME_PREFIX="grd"
+COMMON_NAME=$(echo "${COMMON_NAME_PREFIX}${RANDOM_NAME_SUFFIX}" | tr "[:upper:]" "[:lower:]")
+export COMMON_NAME
+
+### Cluster name must be less than 10 characters!
+export CLUSTER_NAME="${COMMON_NAME}"
+
+# set pipefail to handle right errors from tests
+set -o pipefail
+
+# Install Kyma form latest release
+kyma::get_last_release_version -t "${BOT_GITHUB_TOKEN}"
+
+export KYMA_SOURCE="${kyma_get_last_release_version_return_version:?}"
+log::info "### Reading release version from RELEASE_VERSION file, got: ${KYMA_SOURCE}"
+
+# checks required vars and initializes gcloud/docker if necessary
+gardener::init
+
+# if MACHINE_TYPE is not set then use default one
+gardener::set_machine_type
+
+kyma::install_cli_last_release
+
+# currently only Azure generates overrides, but this may change in the future
+gardener::generate_overrides
+
+log::info "### Provisioning Gardener cluster"
+gardener::provision_cluster
+
+log::info "### Installing Kyma $KYMA_SOURCE"
+
+if [[ "${KYMA_MAJOR_VERSION}" == "2" ]]; then
+  gardener::deploy_kyma
+else
+  gardener::install_kyma
+fi
+
+# generate pod-security-policy list in json
+utils::save_psp_list "${ARTIFACTS}/kyma-psp.json"
+
+# test the default Eventing backend which comes with Kyma
+eventing::pre_upgrade_test_fast_integration_eventing
+
+# upgrade the kyma to the current PR/commit state
+
+# set KYMA_SOURCE used by gardener::install_kyma
+# at the time of writing this comment, kyma-integration-gardener never sets BUILD_TYPE to "release"
+if [[ -n ${PULL_NUMBER} ]]; then
+    # In case of PR, operate on PR number
+    KYMA_SOURCE="PR-${PULL_NUMBER}"
+    export KYMA_SOURCE
+elif [[ "${BUILD_TYPE}" == "release" ]]; then
+    readonly RELEASE_VERSION=$(cat "VERSION")
+    log::info "Reading release version from RELEASE_VERSION file, got: ${RELEASE_VERSION}"
+    KYMA_SOURCE="${RELEASE_VERSION}"
+    export KYMA_SOURCE
+else
+    # Otherwise (master), operate on triggering commit id
+    if [[ -n ${PULL_BASE_SHA} ]]; then
+        readonly COMMIT_ID="${PULL_BASE_SHA::8}"
+        KYMA_SOURCE="${COMMIT_ID}"
+        export KYMA_SOURCE
+    else
+        # periodic job, so default to main
+        KYMA_SOURCE="main"
+        export KYMA_SOURCE
+    fi
+fi
+# uses previously set KYMA_SOURCE
+log::info "### Upgrading Kyma to $KYMA_SOURCE"
+
+if [[ "${KYMA_MAJOR_VERSION}" == "2" ]]; then
+  gardener::deploy_kyma
+else
+  gardener::install_kyma
+fi
+
+# test the eventing fi tests after the upgrade
+eventing::fast_integration_tests
+
```

Comment:
There should be one more step here, where you upgrade/reconcile the cluster (which is now at PR-version) to the same PR-version and then run the tests again and cleanup. This is to make sure that further reconciliations do not break the cluster.

Reply:
it was only for debugging purposes, so I put the post-upgrade tests back